tests/test_middleware.py

```diff
@@ -0,0 +1,38 @@
+from unittest import mock
+
+import pytest
+
+from django.contrib.auth.models import AnonymousUser
+from django.test import TestCase
+from django.test.client import RequestFactory
+
+from authbroker_client.middleware import ProtectAllViewsMiddleware
+
+
+def get_response_fake(request):
+    return True
+
+
+@pytest.mark.django_db
+class AnonymousUserAccessibilityTests(TestCase):
+    def setUp(self):
+        factory = RequestFactory()
+        self.request = factory.get('/')
+        self.request.user = AnonymousUser()
+
+    @mock.patch('authbroker_client.middleware.settings', AUTHBROKER_ANONYMOUS_PATHS=())
+    @mock.patch('authbroker_client.middleware.redirect')
+    def test_no_anonymous_paths(self, redirect, settings):
+        middleware = ProtectAllViewsMiddleware(get_response=get_response_fake)
+
+        middleware(request=self.request)
+        redirect.assert_called_once_with("authbroker_client:login")
+
+    @mock.patch('authbroker_client.middleware.settings', AUTHBROKER_ANONYMOUS_PATHS=("/", ))
+    @mock.patch('authbroker_client.middleware.redirect')
+    def test_anonymous_path(self, redirect, settings):
+        middleware = ProtectAllViewsMiddleware(get_response=get_response_fake)
+
+        outcome = middleware(request=self.request)
```

Comment:
Is it worth testing the status code here?

It could also be worth naming `outcome` `response`

asset response.status_code == 200 maybe?

Reply:
I have updated test_no_anonymous_paths as you suggested, however, the object coming back from the middleware function in test_anonymous_path is a boolean, not a response object and therefore I could not implement your suggestion for this test. It might be enough to simply check the boolean result but I feel checking the redirect function makes the intention behind the test more obvious to other developers.

Reply:
Ah, I see you have:

```def get_response_fake(request):
    return True```

So it'll return True - which whilst not a response object, it demonstrates that the middleware doesn't intercept the request.

This is really minor - maybe it'd make it a bit more self-documenting to return a string instead of True, e.g:

```def get_response_fake(request):
    return "the-public-view"```

then 

assert outcome == "the-public_view"

Other than that it looks good! 👍